homeassistant/components/tuya/cover.py

```diff
@@ -179,25 +241,29 @@ def async_discover_device(device_ids: list[str]) -> None:
 class TuyaCoverEntity(TuyaEntity, CoverEntity):
     """Tuya Cover Device."""
 
-    _current_position: IntegerTypeData | None = None
     _current_state: DPCode | None = None
-    _set_position: IntegerTypeData | None = None
-    _tilt: IntegerTypeData | None = None
-    _motor_reverse_mode_enum: EnumTypeData | None = None
     entity_description: TuyaCoverEntityDescription
 
     def __init__(
         self,
         device: CustomerDevice,
         device_manager: Manager,
         description: TuyaCoverEntityDescription,
+        *,
+        current_position: _DPCodePositionWrapper | None = None,
+        set_position: _DPCodePositionWrapper | None = None,
+        tilt_position: _DPCodePositionWrapper | None = None,
     ) -> None:
         """Init Tuya Cover."""
         super().__init__(device, device_manager)
         self.entity_description = description
         self._attr_unique_id = f"{super().unique_id}{description.key}"
         self._attr_supported_features = CoverEntityFeature(0)
 
+        self._current_position = current_position or set_position
+        self._set_position = set_position
+        self._tilt_position = tilt_position
+
         # Check if this cover is based on a switch or has controls
         if get_dpcode(self.device, description.key):
```

Comment:
Note: I have migrated all the integer types in this PR, but I have kept the control (on/off and open/pause/close) for a follow-up PR as I felt this one was big enough as it was.